### DISCUSSION TITLE
Image upload to external hosting

### DISCUSSION THREAD

**Discussion by fievel** [aeharding/voyager#1050] [Ideas]
**Is your feature request related to a problem? Please describe.**

My instance is now blocking image upload for legal issues due to bots uploading CSAM. Anyway a lot of instances are limiting size of image upload due to obvious financial reasons.

**Describe the solution you'd like**

Could be very nice to implement a configurable external image hosting provider, so the "embed image button" could upload to the configured service instead of the instance if the setting is enabled. 

**Additional context**

After upload to external service, the url of the image should be retrieved and added to the markdown post/comment with syntax `![](url)`.

Example services:

- https://vgy.me/
- https://imgur.com/
- https://lutim.lagout.org/

**Comment by aeharding** [comment]
Hi there!

Voyager will not support image upload from a privately owned/"single instance" upload service for various hopefully obvious reasons.

Instead, if there is a viable self-hostable, open source option that supports CORS, I would be open to adding support for that service's API. If so, please create a new issue with a proposal, and outline the upload API. Thanks!
